lms/djangoapps/grades/tasks.py

```diff
@@ -43,12 +44,15 @@ def recalculate_subsection_grade(user_id, course_id, usage_id):
         set()
     )
 
-    for subsection_usage_key in subsections_to_update:
-        transformed_subsection_structure = get_course_blocks(
-            student,
-            subsection_usage_key,
-            collected_block_structure=collected_block_structure,
-        )
-        subsection_grade_factory.update(
-            transformed_subsection_structure[subsection_usage_key], transformed_subsection_structure
-        )
+    try:
+        for subsection_usage_key in subsections_to_update:
+            transformed_subsection_structure = get_course_blocks(
+                student,
+                subsection_usage_key,
+                collected_block_structure=collected_block_structure,
+            )
+            subsection_grade_factory.update(
+                transformed_subsection_structure[subsection_usage_key], transformed_subsection_structure
+            )
+    except IntegrityError as exc:
+        raise recalculate_subsection_grade.retry(args=[user_id, course_id, usage_id], exc=exc)
```

Comment:
Seems like this is going to be a common pattern.  For these common cases where we're just triggering a retry on certain errors, I'd love to see a decorator we could use with a signature like:

```python
@task_with_retries(retry_on=(IntegrityError, OtherError))
def mytask(args):
     """Leave off the try/except now"""
```